pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -55,10 +86,41 @@ buildDotnetModule (finalAttrs: {
   makeWrapperArgs = [
     "--prefix PATH : ${lib.makeBinPath finalAttrs.runtimeInputs}"
     # Make associating with nxm links work on Linux
-    "--set APPIMAGE ${placeholder "out"}/bin/NexusMods.App"
+    # FIXME: this may no longer be needed, testing without it
+    # "--set APPIMAGE ${placeholder "out"}/bin/NexusMods.App"
   ];
 
-  runtimeInputs = [ desktop-file-utils ];
+  # FIXME: Decide if we should use pname (nexusmods-app) or id (com.nexusmods.app) in filenames
+  # Most desktop entries in nixpkgs seem to use pname, but is that just for historical reasons?
+  # Note: the appstream file lists the "desktop-id" launchable as "com.nexusmods.app.desktop"
+  # Other packages seem to interpolate `pname` in the install paths - should I use `finalAttrs.pname`?
+  postInstall = ''
+    # Desktop entry
+    install -D -m 444 -t $out/share/applications src/NexusMods.App/com.nexusmods.app.desktop
+    substituteInPlace $out/share/applications/com.nexusmods.app.desktop \
+      --replace-fail '${"$"}{INSTALL_EXEC}' "$out/bin/NexusMods.App"
```

Comment:
Reading https://github.com/NixOS/nixpkgs/issues/308324, it seems using an absolute path here is bad practice.

Perhaps `'${INSTALL_EXEC}'` should be replaced with `finalAttrs.meta.mainProgram`, to match `lib.getExe`'s behaviour?